.github/workflows/build_and_test.yaml

```diff
@@ -12,71 +12,46 @@ on:
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./ci/actions/setup-deps
+
+      - run: make lint-deps
+      - run: make -k lint
+
   build-and-test:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - name: Run yamllint
-        run: |
-          MAKE_IN_DOCKER=1 make lint-yamllint
-      - name: Run Codespell
-        uses: codespell-project/actions-codespell@master
-        with:
-          # can't reuse tools/codespell/.codespell.skip, shell is not supported
-          skip: .git,*.png,*.woff,*.woff2,*.eot,*.ttf,*.jpg,*.ico,*.svg,go.mod,go.sum
-          ignore_words_file: 'tools/codespell/.codespell.ignorewords'
-          check_filenames: true
-          check_hidden: true
-      - uses: actions/setup-go@v3
-        with:
-          go-version: ${{ env.GO_VERSION }}
-          cache: true
-      - name: Run golangci-lint
-        uses: golangci/golangci-lint-action@v2
-        with:
-          version: v1.46.2
-          args: --build-tags=e2e
-      - name: Build
-        run: make build-all
-      - name: Test and report coverage
-        run: go test ./... -race -coverprofile=coverage.xml -covermode=atomic
+      - uses: ./ci/actions/setup-deps
+
+      # build
+      - run: make docker-build
+
+      # test
+      - run: go test ./... -race -coverprofile=coverage.xml -covermode=atomic
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@v2
         with:
           fail_ci_if_error: true
           files: ./coverage.xml
           name: codecov-envoy-gateway
           verbose: true
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v2
-      - name: Build
-        uses: docker/build-push-action@v3
-        with:
-          file: ./Dockerfile
-          context: bin
-          platforms: linux/amd64
-          tags: ${{ env.ENVOY_GATEWAY_DEV_IMAGE }}:${{ github.sha }}
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
+
+      # push
       - name: Login to DockerHub
         if: github.event_name == 'push'
         uses: docker/login-action@v2
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_PASSWORD }}
-      - name: Push to envoyproxy/gateway-dev
+      - run: make docker-push
+        if: github.event_name == 'push'
+      - name: Push to :latest
         if: github.event_name == 'push' && github.ref == 'refs/heads/main'
-        uses: docker/build-push-action@v3
-        with:
-          file: ./Dockerfile
-          context: bin
-          platforms: linux/amd64,linux/arm64
-          push: true
-          tags: ${{ env.ENVOY_GATEWAY_DEV_IMAGE }}:${{ github.sha }}, ${{ env.ENVOY_GATEWAY_DEV_IMAGE }}:${{ env.ENVOY_GATEWAY_DEV_TAG }}
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
+        run: |
+          docker tag "docker.io/envoyproxy/gateway-dev:$(git rev-parse --short HEAD)" docker.io/envoyproxy/gateway-dev:latest
```

Comment:
This makes the pushed image to amd64 (CI host arch) from multi arch.

Reply:
I think that's fine for now - we need to have one thing working before we worry about having multi-arch. Currently, we have nothing.